### PR TITLE
Manual changes to conform to google style

### PR DESCRIPTION
**Pull request by elefeint**
## Other changes
* Sonar fixes (debt from last PR, which touched lines Sonar did not approve of. None were actual bugs.)
* Reenable checkstyle verification.

### CODE REVIEW

**Review by meltsufin — APPROVED** [review]
LGTM.
Should we create issues for the disabled checks?

File: spring-cloud-gcp-data-datastore/src/main/java/com/google/cloud/spring/data/datastore/core/mapping/DatastorePersistentEntityImpl.java [hunk 1]
```diff
@@ -76,22 +77,28 @@ public DatastorePersistentEntityImpl(
...
 
+  private String findKindName(Class<?> rawType) {
+    if (this.hasTableName()) {
+      // hasTableName() pre-checks kind; the below assert will never trigger.
+      Assert.notNull(this.kind, "Kind unknown; check your @Entity declaration.");
+      return this.kind.name();
+    }
+    return StringUtils.uncapitalize(rawType.getSimpleName());
+  }
+
```
Comment (elefeint) [inline spring-cloud-gcp-data-datastore/src/main/java/com/google/cloud/spring/data/datastore/core/mapping/DatastorePersistentEntityImpl.java:101]: Sonar NPE smell fix.

File: spring-cloud-gcp-data-datastore/src/main/java/com/google/cloud/spring/data/datastore/core/mapping/EmbeddedType.java [hunk 2]
```diff
@@ -50,20 +50,21 @@ public static EmbeddedType of(TypeInformation typeInformation) {
...
-            && typeInformation
-                .getComponentType()
-                .getType()
-                .isAnnotationPresent(
-                    com.google.cloud.spring.data.datastore.core.mapping.Entity.class))
-        || typeInformation
-            .getType()
-            .isAnnotationPresent(
-                com.google.cloud.spring.data.datastore.core.mapping.Entity.class)) {
+    } else if ((typeInformation.isCollectionLike() && isEntity(typeInformation.getComponentType()))
```
Comment (elefeint) [inline spring-cloud-gcp-data-datastore/src/main/java/com/google/cloud/spring/data/datastore/core/mapping/EmbeddedType.java:53]: Sonar NPE smell fix.

File: spring-cloud-gcp-data-spanner/src/main/java/com/google/cloud/spring/data/spanner/repository/query/SpannerQueryLookupStrategy.java [hunk 3]
```diff
@@ -77,12 +77,11 @@ public RepositoryQuery resolveQuery(
...
-    boolean isDml =
-        queryMethod.getQueryAnnotation() != null && queryMethod.getQueryAnnotation().dmlStatement();
+    Query queryAnnotation = queryMethod.getQueryAnnotation();
+    boolean isDml = queryAnnotation != null && queryAnnotation.dmlStatement();
 
-    if (queryMethod.hasAnnotatedQuery()) {
-      Query query = queryMethod.getQueryAnnotation();
-      return createSqlSpannerQuery(entityType, queryMethod, query.value(), isDml);
+    if (queryAnnotation != null && queryMethod.hasAnnotatedQuery()) {
+      return createSqlSpannerQuery(entityType, queryMethod, queryAnnotation.value(), isDml);
```
Comment (elefeint) [inline spring-cloud-gcp-data-spanner/src/main/java/com/google/cloud/spring/data/spanner/repository/query/SpannerQueryLookupStrategy.java:84]: Sonar NPE smell fix.

File: spring-cloud-gcp-samples/pom.xml [hunk 4]
```diff
@@ -21,19 +21,25 @@
 	<properties>
 		<main.basedir>${basedir}/..</main.basedir>
 
-		<!-- Checkstyle version settings. Keep in sync with ../pom.xml -->
-		<spring-cloud-build-tools.version>3.1.0</spring-cloud-build-tools.version>
```
Comment (elefeint) [inline spring-cloud-gcp-samples/pom.xml:25]: Removed since we don't rely on Spring style anymore.

**Review by meltsufin — APPROVED** [review]

**Pull request merged**